### DISCUSSION TITLE
Best way to get chat messages for a specific video for a chat overlay

### DISCUSSION THREAD

**Discussion by catdevcatdev** [JohnXLivingston/peertube-plugin-livechat#66] [Q&A]
Hi! So I intend on making a chat overlay that can be used to overlay on a stream, for livestreamers in an app such as streamlabs. I was wondering what would be the easiest way to get the chat messages? I could actually just use the popout chat itself, but I just need to add additional styling, specific to the overlay. What would be the easiest way to go about this?

Thanks!

**Comment by JohnXLivingston** [comment]
Hello,

I never did such things, so I don't really know what is possible and how it works (streamlabs, overlays, ...).
You can indeed directly include the webchat window in your tool.

NB: the chat url is like https://www.yiny.org/plugins/livechat/4.0.0/router/webchat/room/399a8d13-d4cf-4ef2-b843-98530a8ccbae , with the plugin version in it. You can also use this url https://www.yiny.org/plugins/livechat/router/webchat/room/399a8d13-d4cf-4ef2-b843-98530a8ccbae , that does not contain the plugin version. So that the url never changes. 
With the plugin v4.0.0, you can choose to group chat rooms by channel. In this case, there is an undocumented url to access the chat for the channel with id X : `your_instance/plugins/livechat/router/webchat/room/channel.X` (where X should be the numerical id of your channel). This link can be found in the settings, by using the button «list rooms».

You can also try to connect to the Prosody server with an XMPP client that can use the BOSH protocol. It should be possible. To find the correct parameters, check the code in this file: https://github.com/JohnXLivingston/peertube-plugin-livechat/blob/main/conversejs/builtin.ts (use the anonymous login, you will not be able to connect easily with a Peertube user). But your client should ignore cross domain warnings/errors.


Perhaps @tytan652 has a solution. He is trying to achieve this by using Matterbridge. See here https://github.com/JohnXLivingston/peertube-plugin-livechat/pull/57 and here https://github.com/JohnXLivingston/peertube-plugin-livechat/issues/58

In the future I plan to add an option to allow XMPP client-2-server and server-2-server connections. This will definitively be a solution. But it is not my priority. And I have some technical problems to solve before I can do this.

**Reply by tytan652** [reply]
> Perhaps @tytan652 has a solution. He is trying to achieve this by using Matterbridge. See here #57 and here #58

[obs-matterbridge-overlay](https://gitlab.com/refrac/obs-matterbridge-overlay/) is now a possible candidate to provide chat overlay from multiple service like Twitch and PeerTube even a Matrix room, the issue for now are:
- Matterbridge 1.22.4 with XMPP anonymous connection feature is not out for now
- For now at least Matterbridge shall be installed on the same server as PeerTube to be able to use the local client-2-server connection
- A maybe not enough fool-proofed documentation for how to install it
